### ISSUE TITLE
App: Terrible performance on macbook air/pro m1

### ISSUE THREAD

**Issue by Loryhoof** [webaverse/app#1836]
Sub 30 FPS in most scenes, 10 FPS in shadows.scn (with post processing)

**Comment by ahadshams** [comment]
i think this is a broader optimisation issue here..
even average gaming rigs have low fps..
we should do some benchmarking on grids.scn with/without different parameters such as light, shadows, post processing etc on different hardware as a start..

**Comment by avaer** [comment]
What does the profile say? Is it some shader?

**Comment by avaer** [comment]
Nobody was looking at this P0 issue so I did.

<img width="1440" alt="Screen Shot 2021-11-22 at 2 32 08 AM" src="https://user-images.githubusercontent.com/6926057/142821214-71bdda83-f6ea-4eb9-9d87-5969658577dc.png">

It is obvious from this profile what the problem is (anyone could have captured this but for some reason didn't...)

Too many bones are being processed, either because additional avatars are running their bone loop, or because we are repeating work in the avatars compute.

**Comment by avaer** [comment]
The above is on a MBA M1, but this profile should be reproable on anything CPU bound.

**Comment by avaer** [comment]
We should fix this issue, but this makes a good case for doing avatar updates in multiplayer by sending bones. I believe this is also how all of the VR multiplayer apps do it, including VRChat.

**Comment by ahadshams** [comment]
yes indeed... we are also conducting detailed profiling and results will be shared with the team as to the different optimisations we have to carry out..

any guidance on this will be helpful

@abeersaqib

**Comment by avaer** [comment]
Additional notes:

- apparently enabling alpha in the gl context is not recommended due to perf, though in our case it is needed for html punch through. It could be disabled unless we need it.
- we could try desynchronized: true
https://github.com/whatwg/html/issues/5466

This is after we fix the avatar bones though.

**Comment by avaer** [comment]
One quick performance pass that would do a lot: https://github.com/webaverse/app/pull/1866
